agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/BaseTracer.java

```diff
@@ -127,7 +131,47 @@ public void addThrowable(final Span span, final Throwable throwable) {
     span.recordException(throwable);
   }
 
+  public void onPeerConnection(final Span span, final InetSocketAddress remoteConnection) {
```

Comment:
`HttpServerTracer` and its subclasses tried to reduce public API as much as possible. This includes hiding all attribute setting behind `startSpan` method. So that end-users would need to call as few methods as possible. Ideally just `startSpan` and `startScope`. Please do the same here.

Pass an object, that represent a connection for a given library, into `startSpan` method and call `onPeerConnection` method (which should become protected) from `startSpan`. Also take a look, if `HttpServerTracer.onConnection` method should be pulled up into `BaseTracer`

Reply:
Ok, probably actual call to this method should happen in `HttpServerTracer.startSpan` and `HttpClientTracer.startSpan`

Reply:
this is only used in Cassandra, DatabaseClientTracer, HttpClientRequestTracingHandler.  is it generic enough to apply to all?

Reply:
Converted it to static for now and a todo #912, will consider to move it to somewhere else.